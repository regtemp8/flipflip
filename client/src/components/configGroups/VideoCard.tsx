import * as React from 'react'

import { Grid, Typography } from '@mui/material'

import { getSourceType, ST } from 'flipflip-common'
import VideoControl from '../player/VideoControl'
import type ChildCallbackHack from '../player/ChildCallbackHack'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import {
  setSceneVideoSpeed,
  setSceneVideoVolume
} from '../../store/scene/slice'
import {
  selectSceneVideoVolume,
  selectSceneVideoSkip
} from '../../store/scene/selectors'

export interface VideoCardProps {
  sceneID: number
  isPlaying: boolean
  mainVideo?: HTMLVideoElement
  mainClipID?: number
  mainClipValue?: number[]
  imagePlayerAdvanceHacks: ChildCallbackHack[][]
}

export default function VideoCard(props: VideoCardProps) {
  const dispatch = useAppDispatch()
  const videoVolume = useAppSelector(selectSceneVideoVolume(props.sceneID))
  const videoSkip = useAppSelector(selectSceneVideoSkip(props.sceneID))

  let mainVideoTitle = ''
  if (props.mainVideo) {
    const source = props.mainVideo.getAttribute('source') as string
    const sourceType = getSourceType(source)
    if (sourceType === ST.video) {
      if (source.startsWith('http')) {
        mainVideoTitle = source.substring(source.lastIndexOf('/') + 1)
      } else {
        mainVideoTitle = source.substring(source.lastIndexOf('\\') + 1)
      }
    } else {
      mainVideoTitle = props.mainVideo.src
      mainVideoTitle = mainVideoTitle.substring(
        mainVideoTitle.lastIndexOf('/') + 1
      )
    }
  }
  return (
    <Grid container spacing={2} alignItems="center">
      <Grid item xs={12}>
        {props.mainVideo && (
          <React.Fragment>
            <Typography>Scene Video:</Typography>
            <Typography variant="body2" style={{ whiteSpace: 'normal' }}>
              {mainVideoTitle}
            </Typography>
            <VideoControl
              player
              useHotkeys
              video={props.mainVideo}
              volume={videoVolume}
              clipID={props.mainClipID}
              clipValue={props.mainClipValue}
              nextTrack={() => {
                props.imagePlayerAdvanceHacks[0][0].fire()
              }}
              onChangeSpeed={(speed: number) =>
                dispatch(
                  setSceneVideoSpeed({ id: props.sceneID, value: speed })
                )
              }
              onChangeVolume={(volume: number) =>
                dispatch(
                  setSceneVideoVolume({ id: props.sceneID, value: volume })
                )
              }
              skip={videoSkip}
            />
          </React.Fragment>
        )}
      </Grid>
    </Grid>
  )
}

;(VideoCard as any).displayName = 'VideoCard'
