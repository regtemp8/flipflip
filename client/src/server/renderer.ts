import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
