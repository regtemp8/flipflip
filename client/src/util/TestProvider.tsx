import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
