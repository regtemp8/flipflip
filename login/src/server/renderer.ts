import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
