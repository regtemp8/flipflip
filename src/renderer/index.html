<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
