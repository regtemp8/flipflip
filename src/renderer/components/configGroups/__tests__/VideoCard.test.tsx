import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import VideoCard from "../VideoCard";
import TestProvider from "../../../../../test/util/TestProvider";

jest.mock('../../player/VideoControl', () => 'VideoControl');

describe("VideoCard", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider>
        <VideoCard
            scene={null}
            otherScenes={[]}
            isPlaying={false}
            mainVideo={null}
            mainClip={null}
            mainClipValue={[]}
            otherVideos={[]}
            imagePlayerAdvanceHacks={[]}
            onUpdateScene={(scene, fn) => {}}
        />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
