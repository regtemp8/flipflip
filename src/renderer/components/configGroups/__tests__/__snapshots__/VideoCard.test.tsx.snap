// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCard should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  />
</div>
`;
